src/hydra/dataProvider.js

```diff
@@ -113,14 +119,36 @@ export const transformJsonLdDocumentToReactAdminDocument = (
  * UPDATE   => PUT http://my.api.url/posts/123
  */
 export default (
-  entrypoint,
+  entrypointOrParams,
   httpClient = fetchHydra,
   apiDocumentationParser = parseHydraDocumentation,
   useEmbedded = false, // remove this parameter for 3.0 (as true)
 ) => {
+  let mercure;
+  let entrypoint = entrypointOrParams;
+  if (typeof entrypointOrParams === 'object') {
+    const params = Object.assign({}, defaultParams, entrypointOrParams);
+    entrypoint = params.entrypoint;
+    httpClient = params.httpClient;
+    apiDocumentationParser = params.apiDocumentationParser;
+    useEmbedded = params.useEmbedded;
+    mercure = Object.assign(
+      {},
+      {
+        hub: `${params.entrypoint}/.well-known/mercure`,
+        jwt: null,
+        topicUrl: window.origin,
+      },
+      params.mercure,
+    );
+  }
+
   /** @type {Api} */
   let apiSchema;
 
+  // store mercure subscriptions
+  const subscriptions = [];
```

Comment:
I think it would be better to use an object here, with the resource ids as keys.